Services/SystemServer/Service.cpp

```diff
@@ -135,7 +135,12 @@ void Service::setup_socket()
     }
 
     auto socket_address = Core::SocketAddress::local(m_socket_path);
-    auto un = socket_address.to_sockaddr_un();
+    auto un_optional = socket_address.to_sockaddr_un();
+    if (!un_optional.has_value()) {
+        perror("bind");
```

Comment:
This is a lie, we're not reporting an error from the `bind` syscall

Reply:
It absolutely is! I was having a hard time coming up with anything.

Now that I think about it more deeply, I notice that it will end up writing `bind: Success` which is arguably even worse than the lie itself.

So it's `dbg()` then?

Reply:
@hard time: I mean how to deal with error propagation across the stack, not only in this particular function.

Reply:
Yeah just `dbg()` :)

Reply:
Switched to dbg().